pkgs/applications/networking/instant-messengers/qq/default.nix

```diff
@@ -67,28 +67,18 @@ stdenv.mkDerivation {
 
   installPhase = ''
     runHook preInstall
-    mkdir -p "$out/share/icons/hicolor/0x0/apps"
-    cp usr/share/icons/hicolor/0x0/apps/qq.png $out/share/icons/hicolor/0x0/apps
 
-    mkdir -p "$out/opt"
-    cp -r "opt/"* $out/opt
+    mkdir -p $out/bin
+    cp -r opt $out/opt
+    cp -r usr/share $out/share
+    substituteInPlace $out/share/applications/qq.desktop \
+      --replace "/opt" "$out/opt" \
+      --replace "/usr/share" "$out/share"
```

Comment:
Could you replace the binary with `$out/bin/qq`? The new desktop file points to the original binary, which doesn't have `wrapGAppsHook`, so it would crash when selecting a file.

Reply:
Done. Thanks!